trade-finance-manager-ui/server/middleware/validatePostAddPaymentRequestBody/index.ts

```diff
@@ -10,23 +10,47 @@ import {
 } from '../../helpers/premium-payments-table-checkbox-id-helper';
 import { PremiumPaymentsTableCheckboxId } from '../../types/premium-payments-table-checkbox-id';
 
+/**
+ * Checks if the given value is an object.
+ * @param body - The value to check.
+ * @returns True if the value is an object, false otherwise.
+ */
 const isRequestBodyAnObject = (body: unknown): body is object => !body || typeof body === 'object';
 
-const FACILITY_ID_QUERY_REGEX = /facilityIdQuery=(?<facilityIdQuery>\d{4,10})/;
-
-const getFacilityIdQueryFromReferer = (req: Request): string | undefined => {
+/**
+ * Regular expression to extract the premium payments facility ID from a URL query string.
+ * The pattern uses {4,10} to match between 4 and 10 digits, ensuring the facility ID
+ * is at least 4 digits long but no more than 10 digits.
+ * The matched digits are captured in a named group 'premiumPaymentsFacilityId'.
+ */
+const PREMIUM_PAYMENTS_FACILITY_ID_QUERY_REGEX = /premiumPaymentsFacilityId=(?<premiumPaymentsFacilityId>\d{4,10})/;
```

Comment:
Are we able to partially use the existing `PARTIAL_FACILITY_ID` regex? Not sure if possible without having a go myself